components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/AssignUserRolesFunctionImpl.java

```diff
@@ -0,0 +1,131 @@
+                    String userStoreDomain = user.getWrapped().getUserStoreDomain();
+                    String username = user.getWrapped().getUserName();
+                    UserRealm userRealm = getUserRealm(tenantDomain);
+                    if (userRealm != null) {
+                        UserStoreManager userStore = getUserStoreManager(tenantDomain, userRealm, userStoreDomain);
+                        if (userStore != null) {
+                            userStore.updateRoleListOfUser(
+                                    username,
+                                    new String[0],
+                                    assigningRoles.toArray(new String[0])
+                            );
+                            return true;
+                        } else {
+                            LOG.debug("Null value user store received for the user "
+                                    + user.getMember("username"));
+                        }
+                    } else {
+                        LOG.debug("Null value user realm received for the user "
+                                + user.getMember("username"));
+                    }
+                } else {
+                    LOG.debug("Null value received while getting wrapped content for user");
+                }
+            } catch (UserStoreException e) {
+                LOG.error("Error in getting user from store at the function ", e);
+            } catch (FrameworkException e) {
+                LOG.error("Error in evaluating the function ", e);
+            }
+        } else {
+            if (user == null) {
+                LOG.error("Invalid value for the user");
+            } else {
+                LOG.error("Invalid value for the new roles");
+            }
+        }
+        return false;
+    }
+
+    private UserRealm getUserRealm(String tenantDomain) throws FrameworkException {
+
+        UserRealm realm;
+        try {
+            realm = AnonymousSessionUtil.getRealmByTenantDomain(UserFunctionsServiceHolder.getInstance()
+                    .getRegistryService(), UserFunctionsServiceHolder.getInstance().getRealmService(), tenantDomain);
+        } catch (CarbonException e) {
+            throw new FrameworkException(
+                    "Error occurred while retrieving the Realm for " + tenantDomain + " to retrieve user roles", e);
+        }
+        return realm;
+    }
+
+    private UserStoreManager getUserStoreManager(String tenantDomain, UserRealm realm, String userDomain)
+            throws FrameworkException {
+
+        UserStoreManager userStore;
+        try {
+            if (StringUtils.isNotBlank(userDomain)) {
+                userStore = realm.getUserStoreManager().getSecondaryUserStoreManager(userDomain);
+            } else {
+                userStore = realm.getUserStoreManager();
+            }
+
+            if (userStore == null) {
+                throw new FrameworkException(
```

Comment:
Why are we throwing FrameworkException? the method which calls the getUserStoreManager already handles UserStoreException.

Reply:
This is the pattern followed in the module

Reply:
let's get it changed, may be with a separate PR for all the modules.